oneview/data_source_logical_interconnect.go

```diff
@@ -0,0 +1,435 @@
+							Optional: true,
+						},
+						"system_contact": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"stacking_health": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"telemetry_configuration": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"category": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"created": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"description": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"eTag": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"enable_telemetry": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"modified": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"sample_count": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"sample_interval": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"state": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"status": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"uri": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"uri": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func dataSourceLogicalInterconnectRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	id := d.Get("name").(string)
+	logInt, err := config.ovClient.GetLogicalInterconnectById(id)
+
+	if err != nil || logInt.URI.IsNil() {
+		d.SetId("")
+		return nil
+	}
+
+	d.SetId(id)
+
+	d.Set("category", logInt.Category)
+	d.Set("consistency_status", logInt.ConsistencyStatus)
+	d.Set("created", logInt.Created)
+	d.Set("description", logInt.Description)
+	d.Set("eTag", logInt.ETAG)
+	d.Set("fusion_domain_uri", logInt.FusionDomainUri)
+	d.Set("logical_interconnect_group_uri", logInt.LogicalInterconnectGroupUri)
+	d.Set("modified", logInt.Modified)
+	d.Set("name", logInt.Name)
+	d.Set("stacking_health", logInt.StackingHealth)
+	d.Set("state", logInt.State)
+	d.Set("status", logInt.Status)
+	d.Set("type", logInt.Type)
+	d.Set("uri", logInt.URI)
```

Comment:
why aren't we setting the other attributes of the resource here?

Reply:
All attributes are added.